main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
As for whether it's necessary or not, I'm ambivalent.  There are probably tens of thousands of `syslog` clients.

If it's the `syslog` server's responsibility to sanitize what goes into log files (and it is, hence Postel's Law), then it should be fixed there.

But I made the mistake of opening a [bug](https://bugs.php.net/bug.php?id=74860) that has precipitated some alarmist and generally unhelpful comments.

I don't want to be accused of throwing caution to the wind and ignoring them, hence the 2nd (and now 3rd) commits.

Frankly I would be happy with just the first one.

If the 2nd and 3rd are deemed necessary, I'll squash those two together before merge.